### DISCUSSION TITLE
How do I SSH into an LXC like Homepage

### DISCUSSION THREAD

**Discussion by mattcollects64** [community-scripts/ProxmoxVE#318] [Q&A] [answered]
I know this is probably elementary, but how do I SSH into my LXCs? I installed homepage for example and when I try to SSH root@IPADDR, it asks for a password. In proxmox, the script autologs me in as root for this LXC in the console, so I don't know the password.

Any advice? Thanks!

Edit: I suppose I could SSH into proxmox itself, then jump down into the container to do my work. Just thought there might be a way to go right to the container.

**Comment by HearthCore** [comment, marked as answer]
LXCs Install scripts have defaults set to NO when it comes to "allow SSH" at all.
Debian has a file here nano /etc/ssh/sshd_config - that has some config, so you can very well allow login directly into an LXC with the LXCs IP adress/hostname and its user:password combination, or limit logging in to keys. (Create keys on your local machine, then copy them to proxmox, then lxc push them to the lxcs you'd want to login without password and keep it locked down somewhat)

**Reply by mattcollects64** [reply]
Thank you very much.

**Reply by netizeni** [reply]
I literally have the same question and found this thread. @HearthCore as I'm not so familiar with this, can you explain a bit more detailed please? Is it possible to find out which password script used when LXC was created, in order to ssh login with user:password combination? 

I found sshd_config in homepage LXC, but I don't know what should I uncomment in order to enable ssh and to login with user:password combination? Tried user/pass combination for proxmox itself, but it didn't work.

I see #PermitRootLogin prohibit-password being commented.

edit: Just to add to this. I would also like to use WinSCP in order to transfer files between LXC and my PC. I assume once I have the password (if it's possible to find it somehow) there won't be a problem to transfer those files.

**Reply by tesla-76** [reply]
@netizeni I do this to enable SSH access to a container, by Proxmox from the CONSOLE of the LXC container:

1 - sudo nano /etc/ssh/sshd_config

2 - uncomment the line: **#PermitRootLogin prohibit-password** and then change it to: **PermitRootLogin yes**

3 - sudo systemctl restart ssh    (restarts the OpenSSH server)

4- sudo passwd    (sets the login password for root user)

The IP of the LXC container can now be accessed via SSH.

P.S. Yes, with SSH access enabled then you can transfer/edit files as well as send commands to the container, I personally use MobaXterm.

**Reply by netizeni** [reply]
@tesla-76 Thank you for detailed explanation!

If I understood correctly, since you change the root password in step 4, does it interfere in any way with Proxmox itself? For example, does console from Proxmox web UI works afterwards, or you have to change the password there somehow as well?

**Reply by tesla-76** [reply]
@netizeni Yes, this step, **if done in the LXC container**, doesn't touch anything in the Proxmox system (to be clear, you're not going to modify the Proxmox login data), it just sets up SSH access via password to that **single LXC container**.

Don't worry that you can't accidentally lock yourself out of Proxmox (I in my tests with 2FA for the main login in Proxmox was managing to lock myself out XD). 

This is a basic system, you can also set access keys or set access privileges for only certain directories, but I am not yet familiar with these methods.

**Comment by Koliera** [comment]
Might be I'm not fully grasping the question, but from my experience I always had a chance to ssh with tailscale installed (also nicer to not require it to be open to the net). Plus, the script to allow tailscale to run in a LXC makes it a breeze to run.

**Reply by Mellowlynx** [reply]
You can always access a container through SSH, If you go through the node itself.
No need to enable SSH or open ports.

It's my preferred way, so the node has my key, and I don't have to manage that in every single LXC.
